### PR TITLE
Cleans up storage.dm

### PR DESCRIPTION
**Pull request by Cyberboss**
I was gonna componentize storage, but then I got lazy. So, have some shitcode cleanup instead.

### CODE REVIEW

File: code/game/objects/items/weapons/storage/storage.dm [hunk 1]
```diff
@@ -9,75 +9,120 @@
...
+/obj/item/weapon/storage/Initialize()
+	. = ..()
+
+	can_hold = typecacheof(can_hold)
+	cant_hold = typecacheof(cant_hold)
+
+	if(allow_quick_empty)
+		verbs += /obj/item/weapon/storage/verb/quick_empty
+	else
+		verbs -= /obj/item/weapon/storage/verb/quick_empty
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/storage/storage.dm]: Only one of these lines should be necessary, but I'm not sure which one...

Comment (MoreRobustThanYou) [inline code/game/objects/items/weapons/storage/storage.dm]: ```verbs += /obj/item/weapon/storage/verb/quick_empty```?

**Comment by Cyberboss** [comment]
Too much work for a no-op PR

**Comment by Cyberboss** [comment]
I'm so fucking lazy jesus fucking christ

File: code/game/objects/items/storage/storage.dm [hunk 2]
```diff
@@ -9,82 +9,118 @@
...
 	icon = 'icons/obj/storage.dmi'
 	w_class = WEIGHT_CLASS_NORMAL
-	var/silent = 0 // No message on putting items in
-	var/list/can_hold = new/list() //Typecache of objects which this item can store (if set, it can't store anything else)
-	var/list/cant_hold = new/list() //Typecache of objects which this item can't store
-	var/list/is_seeing = new/list() //List of mobs which are currently seeing the contents of this item's storage
+	var/silent = FALSE // No message on putting items in
+	var/list/can_hold = list() //Typecache of objects which this item can store (if set, it can't store anything else)
+	var/list/cant_hold = list() //Typecache of objects which this item can't store
+	var/list/is_seeing = list() //List of mobs which are currently seeing the contents of this item's storage
```
Comment (RemieRichards) [inline code/game/objects/items/storage/storage.dm]: assoc pls

File: code/game/objects/items/storage/storage.dm [hunk 3]
```diff
@@ -9,82 +9,118 @@
...
 
-		// this must come before the screen objects only block, dunno why it wasn't before
-		if(over_object == M && M.CanReach(src,view_only = TRUE))
-			orient2hud(M)
-			if(M.s_active)
-				M.s_active.close(M)
-			show_to(M)
-			return
+	boxes = new
+	boxes.name = "storage"
```
Comment (AnturK) [inline code/game/objects/items/storage/storage.dm]: Shouldn't these be just default values on /storage then ?

**Review by ghost — APPROVED** [review]
this approve probably won't let you merge it but you get my kinda-useless seal of approval

**Pull request closed without merging**